### PR TITLE
Rename model to note type in content provider

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
In this case, I only renamed private values and local variables,as the content provider is used to provide content to external entities, its API should not be changed. I also added some documentation to explain why we have those old names.

We could consider duplicating "model" to "note type" in the entire API, and noting "model" as deprecated. But I really don't think that's it's worth deprecating anything. And if we don't plan to deprecate, I don't think it's worth having duplicate API either.

### CODE REVIEW

File: api/src/main/java/com/ichi2/anki/FlashCardsContract.kt [hunk 1]
```diff
@@ -432,7 +435,8 @@ public object FlashCardsContract {
         public const val _ID: String = "_id"
 
         /**
-         * This is the ID of the model that this row belongs to (i.e. [Model._ID]).
+         * This is the ID of the note type that this row belongs to (i.e. [Model._ID]).
+         * "Model" was the previous name of "note type". We keep this deprecated name for the sake of compatibility with the ecosystem.
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/FlashCardsContract.kt:439]: Why not add a `NOTE_TYPE_ID` constant which we prefer people to use?

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 2]
```diff
@@ -87,21 +87,21 @@ class ContentProviderTest : InstrumentedTest() {
      */
     private val testDeckIds: MutableList<Long> = ArrayList(TEST_DECKS.size + 1)
     private lateinit var createdNotes: ArrayList<Uri>
-    private var modelId: Long = 0
+    private var noteTypeId: Long = 0
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:90]: likely `: NoteTypeId`

**Comment by david-allison** [comment]
Commit 2 has a typo in the commit title

File: api/src/main/java/com/ichi2/anki/FlashCardsContract.kt (hunk 1 above)
Comment (Arthur-Milchior) [reply api/src/main/java/com/ichi2/anki/FlashCardsContract.kt:439]: I'm not a fan of having synonym in the codebase. Unless one will be deprecated. And I don't think it's worth actually deprecating anything

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt]: Type is useless here. 0L is sufficient

Comment (david-allison) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt]: It's a typealias, so just for documentation

**Review by lukstbit — APPROVED** [review]

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 3]
```diff
@@ -166,21 +166,21 @@ class ContentProviderTest : InstrumentedTest() {
...
         col.modSchemaNoCheck()
-        removeAllModelsByName(col, BASIC_MODEL_NAME)
-        removeAllModelsByName(col, TEST_MODEL_NAME)
+        removeAllNoteTypeByName(col, BASIC_NOTE_TYPE_NAME)
+        removeAllNoteTypeByName(col, TEST_NOTE_TYPE_NAME)
     }
 
     @Throws(Exception::class)
-    private fun removeAllModelsByName(
+    private fun removeAllNoteTypeByName(
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:176]: Note Types

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 4]
```diff
@@ -552,69 +552,69 @@ class ContentProviderTest : InstrumentedTest() {
...
             for (i in 0 until fields.length()) {
                 assertEquals(
                     "Check name of fields",
-                    TEST_MODEL_FIELDS[i],
+                    TEST_NOTE_TYPE_FIELDS[i],
                     fields[i].name,
                 )
             }
-            // Test updating the model CSS (to test updating MODELS_ID Uri)
+            // Test updating the note type CSS (to test updating MODELS_ID Uri)
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:594]: ID

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt [hunk 5]
```diff
@@ -552,69 +552,69 @@ class ContentProviderTest : InstrumentedTest() {
...
             col = reopenCol()
-            model = col.notetypes.get(mid)
-            assertNotNull("Check model", model)
-            assertEquals("Check css", TEST_MODEL_CSS, model!!.getString("css"))
-            // Update each of the templates in model (to test updating MODELS_ID_TEMPLATES_ID Uri)
-            for (i in TEST_MODEL_CARDS.indices) {
+            noteType = col.notetypes.get(mid)
+            assertNotNull("Check note type", noteType)
+            assertEquals("Check css", TEST_NOTE_TYPE_CSS, noteType!!.getString("css"))
+            // Update each of the templates in note type (to test updating MODELS_ID_TEMPLATES_ID Uri)
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:605]: ID

**Comment by david-allison** [comment]
Force pushed to fix my nitpicks. https://github.com/ankidroid/Anki-Android/labels/Pending%20Merge

Thanks!

**Pull request merged**